UKHomeOffice/nmsw-ui
# NSMW-324 Voyage check your answers routes and skeleton page

# Ticket

NMSW-324

----

## AC

prototype: /mvp-2/check-your-answers

Page | URL | Link type | Link name | Link route | Next page
-- | -- | -- | -- | -- | --
Check answers and submit | /report-voyage/check-your-answers | change link | Change | /report-voyage/upload-crew-details | Crew: upload crew details
Check answers and submit | /report-voyage/check-your-answers | change link | Change | /report-voyage/any-passenger-details | Passenger: any passenger details
Check answers and submit | /report-voyage/check-your-answers | change link | Change | /report-voyage/upload-supporting-documents | Supporting: upload supporting documents
Check answers and submit | /report-voyage/check-your-answers | download saved file | Crew.xlsx (or filename) | stay on page | na - file download
Check answers and submit | /report-voyage/check-your-answers | download saved file | Passenger.xlsx (or filename) | stay on page | na - file download
Check answers and submit | /report-voyage/check-your-answers | primary submit button | Save and submit | /report-voyage/report-submitted | Confirmation of submission
Confirmation of submission | /report-voyage/report-submitted | text link | Return to your voyages | /your-voyages | Voyage home



----

## To test

- run tests, all should pass
- run app
- go to `/report-voyage`
- click on check answers and submit
- > you should be taken to `/report-voyage/check-your-answers`
- click on change next to your voyage
- > you should be taken to `/report-voyage/upload-voyage-details`
- go back to cya page
- click on change next to `crew`
- > you should be taken to `/report-voyage/upload-crew-details`
- go back to cya page
- click on change next to `passengers`
- > you should be taken to `/report-voyage/passenger-details`
- go back to cya page
- click on change next to `supporting documents`
- > you should be taken to `/report-voyage/upload-supporting-documents`
- go back to cya page
- click on submit button
- > you should see a console log with the voyage id (see note)

----

## Notes

- confirmation of submission page not done, you will just get a console log
- hardcoded voyage ID being passed from task list to this page
